### ISSUE TITLE
Feature Request: sixel graphics mode

### ISSUE THREAD

**Issue by BrainDamage** [termux/termux-app#142]
https://en.wikipedia.org/wiki/Sixel is a raster protocol format part of VT200 and VT300 serial terminals
it recently started gaining traction on command line tools to output images directly to the terminal because it works trought ssh, works reasonably well, etc 
for an implementation detail to copy the encoder/decoder, one can look at https://github.com/saitoha/libsixel , the url documents also a series of terminals and programs that support the standard
I realize it's a low priority feature, but it'd allow rather nice semi graphical workflow under termux's terminal, currently  it's already possible to view sixel graphics from launching a ssh daemon and connecting externally with a sixel capable terminal emulator

**Comment by EralpCelebi** [comment]
İs there any development ? I would really like to use this thing.

**Comment by xeffyr** [comment]
This should be good. But can you provide an example of software that supports sixel graphics ?

**Comment by its-pointless** [comment]
Gnuplot does 
I think w3m can or one textmode brower can display sixel images

**Comment by beiriannydd** [comment]
https://devhub.io/repos/saitoha-libsixel says it better than I can.  I would really appreciate this feature.

**Comment by petrmachcz** [comment]
Is it supported now?

**Comment by hackerb9** [comment]
I'm the author of `lsix` which is like ls but shows images in the terminal using SIXEL, https://hackerb9.com/lsix. I have received a request for Termux support. I believe that my program should work fine with Termux once it supports SIXEL graphics.

**Comment by bakaika** [comment]
Neofetch supports sixel, instead of distro art, you can use an image.

**Comment by cogburnd02** [comment]
I would love for this to be implemented. 

Perhaps some code can be ported over from yaft-android?? (MIT licensed)

https://github.com/uobikiemukot/yaft-android

**Comment by Efreak** [comment]
I'm quite interested in using this with neofetch, lsix. This could also be used for previewing formatted documents directly in the terminal.

**Comment by cogburnd02** [comment]
@Efreak RIGHT? A version of fbgs (https://github.com/kraxel/fbida/blob/master/fbgs) modified to use img2sixel instead of fbi would be great for viewing ps and pdf files such as those produced by groff or TeX and wa (https://developer.run/37) could make use of img2sixel to show results from Wolfram Alpha right in the command line!

**Comment by evert-mouw** [comment]
Sixel support would be very nice.

**Comment by nonetrix** [comment]
Would like the idea of being able to use w3m with images

**Comment by evert-mouw** [comment]
Konsole now supports Sixel. Increasing support in many places.

https://invent.kde.org/utilities/konsole/-/merge_requests/594

**Comment by cogburnd02** [comment]
Meant to comment this before, but I suppose 'better late than never,' right?

I put up a bounty for this on bountysource: https://app.bountysource.com/issues/36956941-feature-request-sixel-graphics-mode

**Comment by NickWilde263** [comment]
how the progress of sixel support?

**Comment by agnostic-apollo** [comment]
It can be looked into in future. Also `master` branch supports android 5/6.

https://github.com/termux/termux-app/wiki/Termux-on-android-5-or-6

**Comment by MatanZ** [comment]
I missed that. Anyway, here's a branch of the current master. A few problems fixed, but still needs work: https://github.com/MatanZ/termux-app/commits/sixel2

**Comment by Grimler91** [comment]
@MatanZ this is awesome, thanks for working on this!

Sneakpeek for everyone:

![sixel2](https://user-images.githubusercontent.com/18900601/188279023-db513998-8ab8-4215-a209-454dc6fedad5.png)

**Comment by FluffyFoxUwU** [comment]
![Screenshot_20220903-231638](https://user-images.githubusercontent.com/67861857/188279361-600a3782-e5d1-41d5-917b-4fa4d16af16b.png)

maybe dont try play gif yet :3 (this is that one gif in libsixel repo)

**Comment by FluffyFoxUwU** [comment]
![Screenshot_20220903-231638](https://user-images.githubusercontent.com/67861857/188279361-600a3782-e5d1-41d5-917b-4fa4d16af16b.png)

maybe dont try play gif yet :3 (this is that one gif in libsixel repo)

**Comment by agnostic-apollo** [comment]
Wow, that looks cool. Once bugs are sorted out, let me know and can try merging it with rest of the pull requests.

**Comment by MatanZ** [comment]
So, I learned a bit of Java, and I am a bit closer to submitting a PR. An ew branch, due to some playing with git an gradle:
https://github.com/MatanZ/termux-app/tree/sixel4

I see that that specific GIF does not work in Konsole (or xterm, for that matter), and neither with libsixel or ImageMagick converter. Looking at the sixels themselves, I do not see why.

**Comment by MatanZ** [comment]
The problem is actually in the file. I corrected it using something like this: https://gist.github.com/BigglesZX/4016539?permalink_comment_id=3202586 and it works.

**Comment by FluffyFoxUwU** [comment]
offtopic, why creating new branch everytime instead commit to existing one?

**Comment by Grimler91** [comment]
[This gif from wikipedia](https://upload.wikimedia.org/wikipedia/en/9/9d/Loituma.gif) renders without issues in Konsole (as far as I can tell), but has similar artifacts in termux. Fixing it with python code as suggested above:

```py
import numpy as np
from PIL import Image, ImageSequence
from array2gif import write_gif

img = Image.open('Loituma.gif')
frames = np.array([np.array(frame.copy().convert('RGB').getdata(),dtype=np.uint8).reshape(frame.size[1],frame.size[0],3) for frame in ImageSequence.Iterator(img)]) 

write_gif(frames, 'Loituma-fixed.gif', fps=5)
```

makes it render fine in termux as well, compare the two gifs shown at the beginning (`img2sixel Loituma.gif`) and end (`img2sixel Loituma-fixed.gif`) of this video:

https://user-images.githubusercontent.com/18900601/188305141-921d40fe-a947-490d-b778-1f8d2f96628d.mp4

**Comment by MatanZ** [comment]
This gif does not work for me in konsole (with both img2sixel and convert), showing similar artefacts, but less severe. 

Are you sure you used the exact same sixel encoder in konsole and termux? It might depend on the version of the encoder or any of its libraries. It is best to convert the image to a sixel file and cat the same file in both emulators, or ssh from termux to the system that runs konsole.

**Comment by Grimler91** [comment]
> Are you sure you used the exact same sixel encoder in konsole and termux? It might depend on the version of the encoder or any of its libraries.

Probably not, I just tested to img2sixel the gif in archlinux's vanilla Konsole package, and with its builtin encoder it seems to work

**Comment by HardcodedCat** [comment]
It works! but gifs are being duplicated on the terminal when expanding and collapsing the keyboard

Regular images are being displayed fine

![Screenshot_20220905-194040_Termux](https://user-images.githubusercontent.com/103902727/188518018-37add37b-2ab8-4b8e-9194-866e915cc4e0.png)

**Comment by defencedog** [comment]
I want to use `matplotlib.use('module://sixel')`
https://github.com/jonathf/matplotlib-sixel

**Comment by cogburnd02** [comment]
> I want to use `matplotlib.use('module://sixel')` https://github.com/jonathf/matplotlib-sixel

If you're interested in math stuff with sixel, there's a version of gnuplot that can do sixel graphics here: https://github.com/csdvrx/sixel-gnuplot and a script to pretty-print equations in latex format here: https://github.com/nilqed/sixelSPAD/blob/master/bin/latex2sixel and a preview/screenshot of it here: https://raw.githubusercontent.com/saitoha/libsixel/data/data/latex2sixel.jpg

**Comment by defencedog** [comment]
@cogburnd02 thanks, but insofar this isnt implemented in `termux`?

**Comment by cogburnd02** [comment]
> @cogburnd02 thanks, but insofar this isnt implemented in `termux`?

Not as of yet, but MatanZ is working on a branch (I think that's what it's called) that does support it. There's a pull request here: https://github.com/termux/termux-app/pull/2973

**Comment by yukiqt** [comment]
7 years later, any updates on if/when this will get added?

**Comment by agnostic-apollo** [comment]
Yeah, planned for next release.

**Comment by AnonymouX47** [comment]
Any ETA for that?

**Comment by agnostic-apollo** [comment]
My ETAs end up being wrong mostly, but mainly 2-3 items left on todo list before I push my changes, will merge pulls after that,  so hopefully end of month for release, but then again, could be delayed.

**Comment by AnonymouX47** [comment]
Okay, thanks. The uncertainty is totally understandable by the wau.

**Comment by FluffyFoxUwU** [comment]
curious about the bounty, has @MatanZ  claimed it?

**Comment by AnonymouX47** [comment]
Please, does Termux currently support any _standard/common_ means of identification and version introspection?... Such as:

- `CSI > q` (preferably)
- `TERM_PROGRAM`, `TERM_PROGRAM_VERSION`

If not, this will be required for the iTerm2 protocol as it provides no means for detecting support.

Thanks.

**Comment by MatanZ** [comment]
I don't like this method for feature detection. It means each application needs to maintain a database of terminals and features they support, and this will, generally, lag actual support.

The way to detect iterm2 graphics support should be using ReportCellSize sequence here https://iterm2.com/documentation-escape-codes.html . A terminal emulator that supports iterm2 image protocol should support this. konsole and termix (with my PR) do.

See example here: 
https://github.com/dotcs/xkcd-cli/blob/919e8cc4c41f9bab29934cbc6b9756c1315a0d59/xkcd_cli/iv.py#L410

**Comment by AnonymouX47** [comment]
> It means each application needs to maintain a database of terminals and features they support, and this will, generally, lag actual support.

True.

> A terminal emulator that supports iterm2 image protocol should support this.

Yes, but the problem with this is the "should". The cell size escape sequence is separate from the image protocol itself, which also doesn't specify that the cell size query be implemented by a terminal emulator implementing the image protocol (which could happen since the terminal could already have other means of querying cell size).

> konsole and termux (with my PR) do.

I see, I guess I never even bothered to check... I just always assumed it was only iTerm2 that supported it.
Wezterm also does (just tried it now)... Never tried it on `mintty` or `hterm` (which both support the image protocol, at least to an extent) though 🤔

* * *

All in all, you're right about this. I think I'll change my detection method and hope no terminal emulator goes astray 🥲 (if so, I'll probably use the name/version method as a fallback for such).

Anyways, I believe the name/version introspection is still useful is other scenarios... particularly where the implementation of a common feature differs from other terminal emulators (and is not a bug) e.g does Termux may not support some image formats or animated images (I should've asked about this earlier), or behave differently by design/platform limitation, etc... and special handling may be needed for such differences.

Thanks.

**Comment by demure** [comment]
> > This should be good. But can you provide an example of software that supports sixel graphics ? I know a termplay program (https://github.com/jD91mZM2/termplay) that can play image/video on terminal but what else ?
> 
> vv, vidxel, w3m with sixel support, ranger

And now tmux 3.4 has added support, making termux sixel even more desireable :)

**Comment by Flexximilian** [comment]
> > > This should be good. But can you provide an example of software that supports sixel graphics ? I know a termplay program (https://github.com/jD91mZM2/termplay) that can play image/video on terminal but what else ?
> > 
> > 
> > vv, vidxel, w3m with sixel support, ranger
> 
> And now tmux 3.4 has added support, making termux sixel even more desireable :)

Don't forget Chafa and caca. 

Can this be merged and released? What's still needed to bring this feature?

**Comment by yeetssite** [comment]
This is still open? Is anybody actually working on this? I would very much like to see this implemented

**Comment by TomJo2000** [comment]
It is still open.
As indicated by its status being `Open`.

But no it is not being worked on anymore, seeing as the PR linked to this issue:
- #2973
is complete and awaiting merger, unfortunately GitHub has no status to indicate that.

**Comment by Kreijstal** [comment]
> This is still open? Is anybody actually working on this? I would very much like to see this implemented

just use termux monet no?

**Comment by TomJo2000** [comment]
> just use termux monet no?

That is a 3rd party fork we offer no support or compatibility guarantees for.
Use according to your own judgement.

**Comment by Kreijstal** [comment]
> > just use termux monet no?
> 
> That is a 3rd party fork we offer no support or compatibility guarantees for. Use according to your own judgement.

🤷 as long as it has sixel :D

**Comment by cryonox** [comment]
yet another +1 here, I believe sixel support is gaining momentum and lots of apps nowadays support it natively. heck even windows terminal support it now. very popular directory browser such as `yazi` supports it. `tmux,mpv` are a few of the very useful utilities that support it too and makes working with terminals very much like a desktop-like experience.

**Comment by nathaneltitane** [comment]
+1

**Comment by TomJo2000** [comment]
> +1

Right, as I've mentioned previously in https://github.com/termux/termux-app/pull/2973#issuecomment-2073213387.
> [!NOTE]
Please do not reply to dormant threads unnecessarily.
It notifies everybody involved in the thread.

**Comment by Kreijstal** [comment]
> Any update on this?

monet

**Comment by lfourquaux** [comment]
> monet

Unmaintained, not on f-droid, so not a solution for everyone.

I agree that some kind of status update from the main Termux developers would be welcome, and could reduce the trickle of comments here.

So, let me ask them: what do you think of MatanZ' pull request https://github.com/termux/termux-app/pull/2973 ?  Is it somehow unsatisfactory?  Or are you uncertain about the idea of supporting sixel in Termux?  Or are you too busy for such a large change?  Is there a roadmap with some steps that must be taken first, before sixel support?  Is see two 119 beta versions, would it not make sense to test sixel support in this development line?

(And, of course, thanks!)

**Comment by Kreijstal** [comment]
> > monet
> 
> Unmaintained, not on f-droid, so not a solution for everyone.
> 
> I agree that some kind of status update from the main Termux developers would be welcome, and could reduce the trickle of comments here.
> 
> So, let me ask them: what do you think of MatanZ' pull request [#2973](https://github.com/termux/termux-app/pull/2973) ? Is it somehow unsatisfactory? Or are you uncertain about the idea of supporting sixel in Termux? Or are you too busy for such a large change? Is there a roadmap with some steps that must be taken first, before sixel support? Is see two 119 beta versions, would it not make sense to test sixel support in this development line?
> 
> (And, of course, thanks!)

you are right... alternative the best thing we can do is just patch it and build it ourselves... I know it is not ideal but if we dont want to wait for mantainers to keep up.

**Comment by kumulangzi** [comment]
***@***.***这个的端口号是多少？








---Original---
From: ***@***.***&gt;
Date: Mon, May 12, 2025 18:02 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: Re: [termux/termux-app] Feature Request: sixel graphics mode (#142)


 Kreijstal left a comment (termux/termux-app#142)
  
Any update on this?
  
monet

**Comment by agnostic-apollo** [comment]
The primary reason it hasn't been added is because I have not had the time to test and review the PR, lot of other things have been going on, but I should be able to get to it in coming weeks, it was already planned to be merged before the main 0.119.0 release.

**Comment by kumulangzi** [comment]
392com392vip@gmail.com这个的端口是多少

**Comment by 621625** [comment]
Hi,

Hope you are well.

I would kindly like to voice my support for Sixel on Termux. Even xfce4-terminal supports it now.

**Comment by anotherdoesnm** [comment]
any update of this issue?

**Comment by twaik** [comment]
No activity means no updates.

**Comment by 621625** [comment]
There is no reason why Termux couldn't do this. It should. Sixel creates a lot of use cases and opens up for fun programs to be created in the near and distant future. Sixel seems archaic but it has a lot of backwards compatibility. Unix is arguably archaic too so they're a great match.

**Comment by twaik** [comment]
The main reason of this merge request not being merged is the lack of time of our maintainers. Not because "termux {can,should} not do this".

**Comment by 621625** [comment]
I see. Keep up the good work. Go Termux. 

Do you think God uses Linux?
If you were swimming in a swamp, would Linux be a raft?

**Comment by yeetssite** [comment]
not to add unnecessary comments but i think god probably uses [TempleOS](https://en.wikipedia.org/wiki/TempleOS) lol

> I see. Keep up the good work. Go Termux.
> 
> Do you think God uses Linux? If you were swimming in a swamp, would Linux be a raft?

**Comment by Devon-Scott** [comment]
+1 on requesting support for this. Seems like this would enable in-terminal PDF rendering and remove the need for X11 or an external android app, which would massively improve `.tex` workflows. I can see at least two PR's open regarding this issue, thank you for your work. Please try to merge this soon! 

Edit: because the PR isn't linked directly to this feature request, the second PR I saw was https://github.com/termux/termux-app/pull/5003, and I'm hoping a maintainer can help merge this soon

**Comment by Potato22** [comment]
+1 on this  
I use a headless rpi as one of the gateways to my NAS and rclone setup to my B2 bucket. Not all times I have access to my desktop machines, so having the ability to view image previews on my termux would be really great.

**Comment by 621625** [comment]
Come on Termux you can do Sixel. There are literally millions of use cases +++. The future is... Sixel : )

**Comment by j-lakeman** [comment]
Pls stop spamming!
You're not adding anything relevant/new!

**Comment by Kreijstal** [comment]
I mean we can vibecode sixel nowadays tho

**Comment by sylirre** [comment]
@621625 Of course sixel is indeed useful and log waiting feature. However project development may not be so straightforward like just merging pull requests, especially when development driven by limited human resources.

Maintainers are free to prioritize their own plans for what they want to implement or change. One may agree or disagree with that, but this is how open-source works and in fact what keeps people motivated to work on project.

For about 6 years we have a *highlighted notice* that Termux app looks for maintainers. For now it is maintained by 1 person, but surely that's not enough. There should be a team of 2-5 persons (depending on time they want to spend) dedicated for work on the Termux app + plugins for active development.

<img width="929" height="358" alt="Image" src="https://github.com/user-attachments/assets/8e17aaf7-aae3-472c-a901-9aad4f231cb0" />

Yes, currently we have 20+ maintainers in Termux org. But they all have different work scopes (e.g. some work exclusively on packages) and activity status.

Begging for adding a feature will bring our attention but unlikely to affect the current flow. This approach doesn't resolve the core reason of the situation.

> I mean we can vibecode sixel nowadays tho

@Kreijstal Sure, vibecode is a thing for projects that you don't need to keep consistent in a long term. Great for MVPs and throw-away stuff. On other hand it is overhyped and making long term maintenance difficult.

**Comment by robertkirkman** [comment]
Arguably, I am also a maintainer (normal permissions level which does not have permission to merge PRs) of termux-app, because I have contributed a PR to it to fix a recent problem. However, I have chosen to temporarily limit my scope of contributions to termux-app to critical fixes only and not new features, because there is going to be an update in the future which will fix a lot of things and needs to eventually be merged into the termux-app repository.

**Comment by sylirre** [comment]
> there is going to be an update in the future which will fix a lot of things and needs to eventually be merged into the termux-app repository

Problem is that there is no way to track progress on this big update. There is no "development" or "termux-next" branch. Everything hold by one person, which effectively means Termux app is solo-maintained.

**Comment by Kreijstal** [comment]
> [@Kreijstal](https://github.com/Kreijstal) Sure, vibecode is a thing for projects that you don't need to keep consistent in a long term. Great for MVPs and throw-away stuff. On other hand it is overhyped and making long term maintenance difficult.

true but I can just have it on custom fork and viberebase it when you push, you won't even notice it.

**Comment by yeetssite** [comment]
> > [@Kreijstal](https://github.com/Kreijstal) Sure, vibecode is a thing for projects that you don't need to keep consistent in a long term. Great for MVPs and throw-away stuff. On other hand it is overhyped and making long term maintenance difficult.
> 
> true but I can just have it on custom fork and viberebase it when you push, you won't even notice it.

Yes but there's inherent problems with that. For instance, without consistency, and consistent communication, it can get very hard and overbearing/overwhelming(I think those may just be the same word smh) to maintain and not release glitchy, buggy, broken updates for a large-scale project such as this. I'm sure if termux was just this fun little app coded by some dudes in a basement it'd be different, but its not, its maintainers have to coordinate with eachother, and termux is one the bigger apps when it comes to running linux environments on android devices, so if they released a vibecoded update that _might_ possibly break something, even if that something is small, it could really break something for somebody, and i don't think they want that. But this is just an educated guess, as someone whos both vibecoded and made legit (although failed) attempts to properly maintain something, I can see why they would want to make sure they can safely add the feature without breaking anything else. They also need to fix bugs and keep the app stable while google aggressively tries to make android more like iOS, so they might not have the time to do so.

-Dutch Van Der Linde

**Comment by robertkirkman** [comment]
> termux is one the bigger apps when it comes to running linux environments on android devices, so if they released a vibecoded update that might possibly break something

there are actually now many forks of Termux App that are vibe coded, but not very many people have noticed them or been using them yet.

I'm not _completely_ sure why more people haven't started using them, maybe because they would, like all other unofficial Termux apps, _not_ be recommended for using here in official Termux GitHub and would not be supported in issues, but what I'm assuming is that _if_ one of them ends up being very good, then people will start randomly mentioning it,

but I haven't noticed that happening yet for any of them, so maybe there isn't any particular one that's very good yet.

**Comment by Kreijstal** [comment]
> > termux is one the bigger apps when it comes to running linux environments on android devices, so if they released a vibecoded update that might possibly break something
> 
> there are actually now many forks of Termux App that are vibe coded, but not very many people have noticed them or been using them yet.
> 
> I'm not _completely_ sure why more people haven't started using them, maybe because they would, like all other unofficial Termux apps, _not_ be recommended for using here in official Termux GitHub and would not be supported in issues, but what I'm assuming is that _if_ one of them ends up being very good, then people will start randomly mentioning it,
> 
> but I haven't noticed that happening yet for any of them, so maybe there isn't any particular one that's very good yet.

can you name at least 3

**Comment by sylirre** [comment]
> can you name at least 3

We don't track forks but they exist

* https://github.com/thejaustin/termux-ai-app
* https://github.com/WangChengYeh/termux_AI
* https://github.com/k-l-lambda/anthroid
* https://github.com/tennox/termux-nix-monet

There are much more than these, including closed source ones

**Comment by lapisxalpha** [comment]
Perhaps this will light a fire somewhere useful? :-)

How about something like THIS as a reason to integrate sixel support:

Compiling Xorg to render with Sixels inside a terminal 
https://github.com/csdvrx/xsixel

That is a fork of the original project. Unfortunately they only have x86-64 binaries released. Anyone (please) interested in porting that to an aarch64 static binary??? Looks promising. (and fun)

**Comment by twaik** [comment]
You know that there is termux-x11, right? So you do not need to play with other X servers unless it does not fulfill your needs completely.

**Comment by robertkirkman** [comment]
> You know that there is termux-x11, right? So you do not need to play with other X servers unless it does not fulfill your needs completely.

Yes but, Termux:X11 has to be inside of a popup window, overlay window, PiP window or split view window, but Xsixel could be inside of a sixel-enabled Termux app itself and be showing directly in the terminal without a separate app, however, I feel that input handling surely can't be very convenient in Xsixel. How can it track cursor inputs in any usable way? Additionally, sixel images are very slow at high resolutions so how can it have a good framerate?

**Comment by 621625** [comment]
Hi gays,

Listen I am NOT the dev of the project. I'm a user that has posted here previously. But I think that new brow6el sixel browser (google it) is a VERY compelling reason to ensure Termux can do sixel finally.

#LostInLasVegas

**Comment by cryonox** [comment]
ok guys. you know instead of posting here if you just open claude/codex whatever you know you will have the feature in an hour right? just in case you are living a well don't know what's going on in the world. software is so trivially accessible now more than ever. if you are not even willing to spend an hour just being beta tester for AI (ya literally), you really sure you need the feature?

**Comment by sylirre** [comment]
@cryonox The only blocker here is a *human factor*

1. Termux app repo is solo-maintained.
2. There is ongoing ~mythical~ rewrite of Termux app, without progress tracking and estimate. A primary reason why it is so hard for pull requests to get merged into termux-app.
3. There is a Sixel pull request ready to be merged: https://github.com/termux/termux-app/pull/2973. But I suppose (2) gets in the way.

AI in termux-app is out of question - may raise concerns about code quality, licensing, etc from specific Termux org members.

I'm just stating a fact about current condition of the public repo. That's not something I can change, particularly because I'm not a maintainer of termux-app (read-only permissions) and unknown status of all pending works.

**Comment by lapisxalpha** [comment]
> > You know that there is termux-x11, right? So you do not need to play with other X servers unless it does not fulfill your needs completely.
> 
> Yes but, Termux:X11 has to be inside of a popup window, overlay window, PiP window or split view window, but Xsixel could be inside of a sixel-enabled Termux app itself and be showing directly in the terminal without a separate app, however, I feel that input handling surely can't be very convenient in Xsixel. How can it track cursor inputs in any usable way? Additionally, sixel images are very slow at high resolutions so how can it have a good framerate?

The author straight up says it's not going to be playing youtube videos... but it might allow web browsing or using a text editor with fonts. The reason we do or try anything that pushes new boundaries is "because we can". Once a new frontier is opened others will find new uses for it. I am biased towards creative uses of the terminal that doesn't need yet another android app. For the same reason that authors created TwIN (Terminal mode WINdowing Environment... see Cosmos72), or VTM when tmux already exists and "works", Maybe Xsixel could be the next evolution but far more flexible built on existing standards. I have absolutely no idea how the performance would scale with cpu/gpu and bandwidth but eventually it might catch up.

It might work over ssh, and Xsixel might even potentially work in termux-float, which I find indespensible despite having only one window and no tabs..its the only game in town (so I use vtm IN termux-float). As for Termux-X11, I really think it would be awesome if the secret update or any other app had floating x-objects or windows that floated above the rest of android's base layer, a-la "Floating Apps". Yes, I could just go use a linux box, but where's the fun in that? The challenges are always at the fringe.

Thanks, everyone for replying. Peace.

AND, if it hasn't been said enough, thank you very much to the solo maintainer and the authors of Termux, and everyone involved in the project for providing something so useful and inspiring to so many, including me. Termux itself seems conceived to push established boundaries and do something noone else had before. No posix? No problem! (well, maybe a few :-)

**Comment by cryonox** [comment]
@sylirre , you don't understand what I meant at all. I understand the human factor perfectly. I also understand why there's a need to get human review when you are using AI-assisted coding. Some people maybe produce shit code, some may produce awesome code. But you don't know which one you are getting, and thus there's a need to review. But that's from the perspective of repo maintaintence and continual contribution ya?  I am not downplaying the importance of maintainer at all. After all it's a voluntary thankless job and nobody has infinite time.

 I am saying, as the code generation tools are so easilly accessible now,  everyone can have their own implementation of this feature "without contributing back to the source". Yes it maybe terrible and full of edge case bugs but it will work well enough. If you are patient enough to go through all the bugs, you might end up something close to perfect. But since you are not reading the code at all, code quality is probably gonna be shit. Hey but it will work! Hopefully nothing a future model can't fix! You can even point the AI at existing PR if you feel like you can't prompt and it will build it for you! 

So would you rather have something that works now? Or wait for some indeterminate future? I am talking to the people that may not have thought of this.

**Comment by robertkirkman** [comment]
> you can even point the AI at existing PR if you feel like you can't prompt and it will build it for you!

Yes build this PR and use it

- https://github.com/termux/termux-app/pull/2973
